File: sdk/java/src/main/java/com/zorroa/zmlp/sdk/domain/Asset/AssetSearch.java
package com.zorroa.zmlp.sdk.domain.Asset;

import java.util.HashMap;
import java.util.Map;

/**
 * Stores an Asset search, is modeled after an ElasticSearch request.
 */
public class AssetSearch {

    /**
     * The search to execute
     */
    private Map<String, Object> search;

    /**
     * A query to execute on nested elements.
     */
    private Map<String, Object> elementQuery;

    public AssetSearch() {
        search = new HashMap();
        elementQuery = new HashMap();
    }

    public AssetSearch(Map<String, Object> search, Map<String, Object> elementQuery) {
        this.search = search;
        this.elementQuery = elementQuery;
    }

    public AssetSearch(Map<String, Object> search) {
        this.search = search;
        this.elementQuery = new HashMap();
    }

    public Map<String, Object> getSearch() {
        return search;
    }

    public void setSearch(Map<String, Object> search) {
        this.search = search;
    }

    public Map<String, Object> getElementQuery() {
        return elementQuery;
    }

    public void setElementQuery(Map<String, Object> elementQuery) {
        this.elementQuery = elementQuery;
    }
}

File: services/archivist/archivist/src/main/kotlin/com/zorroa/archivist/domain/AssetSearch.kt
package com.zorroa.archivist.domain

import com.fasterxml.jackson.core.type.TypeReference
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

@ApiModel("Asset Search", description = "Stores an Asset search, is modeled after an ElasticSearch request.")
class AssetSearch(

    @ApiModelProperty(
        "The search to execute",
        reference = "https://www.elastic.co/guide/en/elasticsearch/reference/7.5/query-filter-context.html"
    )
    val search: Map<String, Any>? = null
)

/**
 * A SimilarityFilter describes a hash and a weight.
 */
class SimilarityFilter(
    val hash: String,
    val weight: Float = 1.0f
) {
    companion object {

        val JSON_MAP_OF: TypeReference<Map<String, List<SimilarityFilter>>> =
            object : TypeReference<Map<String, List<SimilarityFilter>>>() {}
    }
}
